src/component.js

```diff
@@ -37,6 +36,17 @@ function registerComponentMagicMethod() {
             subtree: true,
         })
 
+        // This only runs after Alpine is fully initialized, and is needed for when
+        // $parent is called **only** from within an event listener and not elsewhere.
+		if (parentComponent.__x && $el.__x) {
+			$el.__x.unobservedData.$nextTick(() => {
+				Object.entries($el.__x.unobservedData.$parent).forEach(value => {
+					parentComponent.__x.$data[value[0]] = value[1]
+				})
+                parentComponent.__x.updateElements(parentComponent)
+			})
+        }
+
```

Comment:
I'm not sure we need all this code. The difference between parent and component is that in the latter you return the proxy, in the former you return the original dataset.

If you change `return data` to return `$el.$parent` it should work.

Reply:
Looks good to me!